ts/receiver/configMessage.ts

```diff
@@ -53,18 +54,22 @@ async function handleGroupsAndContactsFromConfigMessage(
   envelope: EnvelopePlus,
   configMessage: SignalService.ConfigurationMessage
 ) {
+  await createOrUpdateItem({
+    id: 'hasSyncedInitialConfigurationItem',
+    value: true,
+  });
+
```

Comment:
this is not the right line for this. It should be right before `const numberClosedGroup = configMessage.closedGroups`
Right now you are always setting this so the 

```const didWeHandleAConfigurationMessageAlready =
    (await getItemById(hasSyncedInitialConfigurationItem))?.value || false;
``` is always true